rust/sbp/src/link.rs

```diff
@@ -351,6 +351,7 @@ mod tests {
         assert_eq!(*count.borrow(), 1);
     }
 
+    #[allow(dead_code)]
```

Comment:
@pcrumley What do you think of this change I had to make in order to have CI pass? See https://snav.slack.com/archives/C05HP6S1D2Q/p1711097571216779 for more context

Reply:
Hey @IsakTjernberg I was on PTO and just getting back into the swing of things. I think it makes more sense to fix the lints like this https://github.com/swift-nav/libsbp/pull/1408 because #[allow(dead_code)] is kind of broad lint. That said, I don't feel strongly